metadata-ingestion/src/datahub/ingestion/source/delta_lake/source.py

```diff
@@ -126,26 +139,75 @@ def create(cls, config_dict: dict, ctx: PipelineContext) -> "Source":
         config = DeltaLakeSourceConfig.parse_obj(config_dict)
         return cls(config, ctx)
 
+    inner_data_type = ""
+
+    def _inner_field_details(self, raw_field, parsed_struct=""):
+        if raw_field.get("type") == "array":
+            if isinstance(raw_field.get("elementType"), dict):
+                return (
+                    str(raw_field.get("elementType").get("type"))
+                    + "<"
+                    + self._inner_field_details(raw_field.get("elementType"))
+                    + ">"
+                )
+            else:
+                return str(raw_field.get("elementType"))
+        elif raw_field.get("type") == "struct":
+            for field in raw_field.get("fields"):
+                if isinstance(field.get("type"), dict):
+                    parsed_struct += (
+                        "{0}:".format(field.get("name"))
+                        if parsed_struct
+                        else "" + "{0}:".format(field.get("name"))
+                    )
+                    parsed_struct += field.get("type").get("type") + "<"
+                    return self._inner_field_details(field.get("type"), parsed_struct)
+                else:
+                    parsed_struct = (
+                            parsed_struct
+                            + "{0}:{1}".format(field.get("name"), field.get("type"))
+                            + ","
```

Comment:
we would not need this comma if this is last field in structure.

Reply:
yes, we are removing that after the loop is finish